### ISSUE TITLE
Combiner has same task name as Reducer

### ISSUE THREAD

**Issue by rmetzger** [stratosphere/stratosphere#300]
I have job with two consecutive mappers. 

```
              Map2
             /
Input -- Map1
             \
              Reducer
```

It seems that Map1 has the name of Reducer. So the name of Reducer occurs twice.
The webinterface does not show that Map1 and Map2 are connected to each other. It looks like Input is connected to map2.
I'm not 100% sure about this bug but it seems weird.

**Comment by fhueske** [comment]
Which webinterface are you talking about (job submission / optimization or execution)?
Can you provide a screenshot?

Some notes about chaining: 
- Output that is going to more than one task cannot be chained (so neither Reducer nor Map2 cannot be chained behind Map1).
- Chaining is an aspect of execution and chained tasks do not have names. They run under the name of their "parent" task.

**Comment by rmetzger** [comment]
Okay, than it's probably not related to chaining.

Its the "Stratosphere Management Console" ;) (the job manager web interface).
I can not really provide a screenshot, because telekom (NDA).

**Comment by fhueske** [comment]
I think, a combiner gets the same name as the reducer. 
Usually, combine tasks are invisible during execution, since they are chained behind the preceding task.
Since, the output is split, the Combiner cannot be chained behind Map1 and runs in an additional task.
Therefore, Map2 would be connected to Input (since, Map1 is probably the Combiner).

So the real mapping might be:
Input -> Map1
Map1 -> Combiner
Map2 -> Map2
Reducer -> Reducer

Would that explain the effect?

**Comment by rmetzger** [comment]
Yes, this is very likely, since I just made my reducer combinable.

Is it possible to automatically add something like "(Combiner)" or so to the name?

**Comment by fhueske** [comment]
Shouldn't be a problem. Will prepare a pull request.

**Comment by fhueske** [comment]
Well... It's not so super easy as I thought.
But we need a more flexible task naming mechanism soon anyways...

**Comment by fhueske** [comment]
Has been fixed in #331
